### PR TITLE
Migrate roles command

### PR DESCRIPTION
**Pull request by RainbowDashLabs**

### CODE REVIEW

File: src/main/java/de/chojo/repbot/commands/roles/Roles.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                        .handler(new RemoveReceiver(guilds))
+                        .argument(Argument.role("role", "command.roles.removereceiver.role.description").asRequired()))
+                .subCommand(SubCommand.of("refresh", "command.roles.refresh.description")
+                        .handler(refresh))
+                .subCommand(SubCommand.of("list", "command.roles.list.description")
+                        .handler(new List(guilds)))
+                .subCommand(SubCommand.of("stackroles", "command.roles.stackroles.description")
+                        .handler(new StackRoles(guilds))
+                        .argument(Argument.bool("stack", "command.roles.stackroles.stack.description")
+                        )
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/Roles.java:60]:
```suggestion
                        .argument(Argument.bool("stack", "command.roles.stackroles.stack.description"))
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/Add.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import de.chojo.jdautil.interactions.slash.structure.handler.SlashHandler;
+import de.chojo.jdautil.localization.util.Replacement;
+import de.chojo.jdautil.wrapper.EventContext;
+import de.chojo.repbot.dao.provider.Guilds;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.util.Collections;
+
+public class Add  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/Add.java:11]:
```suggestion
public class Add implements SlashHandler {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/AddDonor.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import de.chojo.jdautil.interactions.slash.structure.handler.SlashHandler;
+import de.chojo.jdautil.localization.util.Replacement;
+import de.chojo.jdautil.wrapper.EventContext;
+import de.chojo.repbot.dao.provider.Guilds;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.util.Collections;
+
+public class AddDonor  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/AddDonor.java:11]:
```suggestion
public class AddDonor implements SlashHandler {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/AddReceiver.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import de.chojo.jdautil.interactions.slash.structure.handler.SlashHandler;
+import de.chojo.jdautil.localization.util.Replacement;
+import de.chojo.jdautil.wrapper.EventContext;
+import de.chojo.repbot.dao.provider.Guilds;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.util.Collections;
+
+public class AddReceiver  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/AddReceiver.java:11]:
```suggestion
public class AddReceiver implements SlashHandler {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/List.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.IMentionable;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+
+public class List  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/List.java:16]:
```suggestion
public class List implements SlashHandler {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/Refresh.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.Set;
+
+import static de.chojo.repbot.util.Guilds.prettyName;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class Refresh  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/Refresh.java:21]:
```suggestion
public class Refresh implements SlashHandler {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/Refresh.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    private static final Logger log = getLogger(Refresh.class);
+    private final RoleAssigner roleAssigner;
+    private final Set<Long> running = new HashSet<>();
+    public Refresh(RoleAssigner roleAssigner) {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/Refresh.java:25]:
```suggestion
    private final Set<Long> running = new HashSet<>();

    public Refresh(RoleAssigner roleAssigner) {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/Remove.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+import de.chojo.jdautil.interactions.slash.structure.handler.SlashHandler;
+import de.chojo.jdautil.localization.util.Replacement;
+import de.chojo.jdautil.wrapper.EventContext;
+import de.chojo.repbot.dao.provider.Guilds;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.util.Collections;
+
+public class Remove  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/Remove.java:11]:
```suggestion
public class Remove implements SlashHandler {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/RemoveDonor.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+import de.chojo.jdautil.interactions.slash.structure.handler.SlashHandler;
+import de.chojo.jdautil.localization.util.Replacement;
+import de.chojo.jdautil.wrapper.EventContext;
+import de.chojo.repbot.dao.provider.Guilds;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.util.Collections;
+
+public class RemoveDonor  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/RemoveDonor.java:11]:
```suggestion
public class RemoveDonor implements SlashHandler {
```

File: src/main/java/de/chojo/repbot/commands/roles/handler/RemoveReceiver.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+import de.chojo.jdautil.interactions.slash.structure.handler.SlashHandler;
+import de.chojo.jdautil.localization.util.Replacement;
+import de.chojo.jdautil.wrapper.EventContext;
+import de.chojo.repbot.dao.provider.Guilds;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.util.Collections;
+
+public class RemoveReceiver  implements SlashHandler {
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/roles/handler/RemoveReceiver.java:11]:
```suggestion
public class RemoveReceiver implements SlashHandler {
```

File: src/main/resources/locale_de_DE.properties [hunk 11]
```diff
@@ -217,36 +217,40 @@ command.repsettings.info.message.title=Reputationseinstellungen
...
+command.roles.removedonor.description=Entfernt eine Spender-Rolle.
+command.roles.removedonor.role.description=Die Spenderrolle, die entfernt werden soll.
+command.roles.removereceiver.description=Entfernt eine Empfänger-Rolle.
+command.roles.removereceiver.message.remove=Empfänger-Rolle %ROLE% entfernt.
+command.roles.removereceiver.role.description=Die Empfängerrolle, die entfernt werden soll.
+command.roles.removeronor.message.remove=Spender-Rolle %ROLE% entfernt.
+command.roles.stackroles.description=Erlaubt das Stapeln von Rollen
+command.roles.stackroles.message.notStacked=Nutzer werden nur die Rolle mit der höchsten Reputation haben.
+command.roles.stackroles.message.stacked=Rollen werden gestapelt. Nutzer haben alle Rollen, die weniger Reputation benötigen, als sie haben.
+command.roles.stackroles.stack.description=Die Rolle, die gestapelt werden soll, wird auf "true" gesetzt.
```
Comment (Taucher2003) [inline src/main/resources/locale_de_DE.properties:253]: Wow, sometimes we just have total garbage in our german translation :joy:

File: src/main/resources/locale_fr_FR.properties [hunk 12]
```diff
@@ -217,36 +217,40 @@ command.repsettings.info.message.title=Paramètres de réputation
...
+command.roles.removedonor.description=Supprimer un rôle de donneur
+command.roles.removedonor.role.description=Le rôle de donateur à supprimer.
+command.roles.removereceiver.description=Supprimer un rôle de récepteur
+command.roles.removereceiver.message.remove=Suppression du rôle de récepteur %ROLE%.
+command.roles.removereceiver.role.description=Le rôle de récepteur à supprimer.
+command.roles.removeronor.message.remove=Suppression du rôle de donneur %ROLE%
+command.roles.stackroles.description=Autoriser le cumul des rôles
+command.roles.stackroles.message.notStacked=L'utilisateur n'aura que le rôle le plus proche de sa réputation.
+command.roles.stackroles.message.stacked=Les rôles seront empilés. Les utilisateurs auront tous les rôles avec moins que leur réputation.
+command.roles.stackroles.stack.description=Set to "true" to stack.
```
Comment (Taucher2003) [inline src/main/resources/locale_fr_FR.properties:253]: Looks like we had missed this missing translation

DeepL suggestion:
```suggestion
command.roles.stackroles.stack.description=Mettre à "true" pour empiler.
```

File: src/main/resources/locale_de_DE.properties (hunk 11 above)
Comment (RainbowDashLabs) [reply src/main/resources/locale_de_DE.properties:253]: Thats why I love it!

**Review by Taucher2003 — APPROVED** [review]

**Pull request merged**